### PR TITLE
REN-42: Remove user ID requirement on post creation

### PR DESCRIPTION
**Pull request by beeduck**
Created security utility for Rent API
Added authorization roles to OAuth2 clients
Changed all usage of 'userPosts' to 'listings'
Removed requirement of sending a user ID to create listing

### CODE REVIEW

File: Rent/Rent-Data/src/main/java/com/rent/data/dataaccess/api/dao/listing/UserPostDAO.java [hunk 1]
```diff
@@ -1,7 +1,7 @@
 package com.rent.data.dataaccess.api.dao.listing;
```
Comment (AsadA128) [inline Rent/Rent-Data/src/main/java/com/rent/data/dataaccess/api/dao/listing/UserPostDAO.java]: Rename to ListingDAO

File: Rent/Rent-Data/src/main/java/com/rent/data/dataaccess/api/dao/listing/UserPostDAOImpl.java [hunk 2]
```diff
@@ -1,8 +1,8 @@
 package com.rent.data.dataaccess.api.dao.listing;
```
Comment (AsadA128) [inline Rent/Rent-Data/src/main/java/com/rent/data/dataaccess/api/dao/listing/UserPostDAOImpl.java:1]: Rename to ListingDAOImpl

**Review by AsadA128 — COMMENTED** [review]
fix it then ill merge

File: Rent/Rent-Data/src/main/java/com/rent/data/dataaccess/api/dao/listing/UserPostDAO.java (hunk 1 above)
Comment (beeduck) [reply Rent/Rent-Data/src/main/java/com/rent/data/dataaccess/api/dao/listing/UserPostDAO.java]: Fixed

File: Rent/Rent-Data/src/main/java/com/rent/data/dataaccess/api/dao/listing/UserPostDAOImpl.java (hunk 2 above)
Comment (beeduck) [reply Rent/Rent-Data/src/main/java/com/rent/data/dataaccess/api/dao/listing/UserPostDAOImpl.java:1]: Fixed

**Review by AsadA128 — APPROVED** [review]
Approved

**Pull request merged**